src/VisualStudio/Core/Def/Implementation/SymbolSearch/RoslynSymbolResult.cs

```diff
@@ -0,0 +1,213 @@
+
+            var classifiedText = new ClassifiedTextElement(excerptResult.ClassifiedSpans.Select(cspan =>
+                new ClassifiedTextRun(cspan.ClassificationType, sourceText.ToString(cspan.TextSpan))));
+            result.ClassifiedContext = classifiedText;
+            /*result.ClassifiedContext = new ClassifiedTextElement(
+                excerptResult.ClassifiedSpans.Select(n => new ClassifiedTextRun(n.ClassificationType, n.
+                );
+            excerptResult.ClassifiedSpans*/
+            // decorate the parent definition:
+            //result.ClassifiedDefinition = await ProtocolConversions.DocumentSpanToLocationWithTextAsync(definition.SourceSpans.First(), sourceText, token).ConfigureAwait(false);
+            result.DefinitionIcon = definition.Tags.GetFirstGlyph().GetImageId();
+            /*
+            // do this for each reference:
+            var classifiedSpansAndHighlightSpan = await ClassifiedSpansAndHighlightSpanFactory.ClassifyAsync(reference.SourceSpan, context.CancellationToken).ConfigureAwait(false);
+            var classifiedSpans = classifiedSpansAndHighlightSpan.ClassifiedSpans;
+            var referenceLocation = await ProtocolConversions.DocumentSpanToLocationAsync(reference.SourceSpan, cancellationToken).ConfigureAwait(false);
+            var docText = await reference.SourceSpan.Document.GetTextAsync(context.CancellationToken).ConfigureAwait(false);
+            var classifiedText = new ClassifiedTextElement(classifiedSpans.Select(cspan => new ClassifiedTextRun(cspan.ClassificationType, docText.ToString(cspan.TextSpan))));
+            var locationWithText = new LSP.LocationWithText { Range = referenceLocation.Range, Uri = referenceLocation.Uri, Text = classifiedText };
+
+            result.ClassifiedContext = 
+            */
+            var mappedDocumentSpan = await TryMapAndGetFirstAsync(documentSpan, sourceText, token).ConfigureAwait(false);
+            if (mappedDocumentSpan.HasValue)
+            {
+                var location = mappedDocumentSpan.Value;
+                result.PersistentSpan = symbolSource.ServiceProvider.PersistentSpanFactory.Create(
+                    documentSpan.Document.FilePath,
+                    location.LinePositionSpan.Start.Line,
+                    location.LinePositionSpan.Start.Character,
+                    location.LinePositionSpan.End.Line,
+                    location.LinePositionSpan.End.Character,
+                    SpanTrackingMode.EdgeInclusive);
+            }
+
+            return result;
+        }
+
+        internal static async Task<RoslynSymbolResult> MakeAsync(RoslynSymbolSource symbolSource, SourceReferenceItem reference, DocumentSpan documentSpan, CancellationToken token)
+        {
+            // TODO: Understand how Roslyn calls GetReferenceGroupsAsync
+            // it seems that at this time, we have full knowledge of definitions
+            // and their references
```

Comment:
Please either resolve a TODO or create a GitHub issue for it.

Reply:
it will be resolved prior to checking in